docs/website/docs/plus/getting-started/transformations.md

```diff
@@ -0,0 +1,446 @@
+
+You can control the lineage mode with the `lineage_mode` parameter on the dlt.transform decorator, which can be set to `strict`, `best_effort`, or `disabled`. The default is `best_effort`, which will try to trace the lineage for each column and set the derived hints but will not fail if this is not possible. SQL-based transformations that introduce non-traceable columns will require you to supply column type hints for any columns that could not be traced. For Python-based transformations, the column type can be derived from the incoming data. In `strict` mode, any transformation that introduces non-traceable columns will fail with a useful error. This mode is recommended if you must ensure which columns may contain sensitive data and need to prevent them from being unmarked. You can also disable lineage for a specific transformation by setting the `lineage_mode` to `disabled`. In this mode, the full schema will be derived from the incoming data and no additional hints will be added to the resulting dataset.
+
+### Untraceable columns
+
+In a couple of scenarios the current implementation is not able to trace the lineage of a column. In some cases you may opt to add a static value to an expression, which is not traceable by definitions. We are planning to allow to add exceptions to the lineage tracing for cases like these. Another case is where a column is the result of some operation on multiple source columns, such as concatenating the value of two separate columns or mathematical operations that use more than one source column. These cases are still being worked on and will be supported in future versions of dlt+.
+
+### Some examples from our company dataset
+
+```python
+
+# the following transformation will fail because the static_column is not traceable
+@dlt_plus.transform(transformation_type="python", lineage_mode="strict")
+def mutated_transformation(dataset: SupportsReadableDataset):
+    return dataset.employees.mutate(static_column=1234)
+
+# we can switch it to best effort mode, which will work
+@dlt_plus.transform(transformation_type="python", lineage_mode="best_effort")
+def mutated_transformation(dataset: SupportsReadableDataset):
+    return dataset.employees.mutate(static_column=1234)
+
+# if we use best effort but switch the transformation type to sql, it will fail
+@dlt_plus.transform(transformation_type="sql", lineage_mode="best_effort")
+def mutated_transformation(dataset: SupportsReadableDataset):
+    return dataset.employees.mutate(static_column=1234)
+
+# we can supply a column hint for the static_column, which will work for sql transformations
+@dlt_plus.transform(transformation_type="sql", lineage_mode="best_effort", columns={"static_column": {"type": "double"}})
+def mutated_transformation(dataset: SupportsReadableDataset):
+    return dataset.employees.mutate(static_column=1234)
+
+```
+
+## SQL vs Python transformations
```

Comment:
Tbh I don't fully understand when you would use one over the other apart from speed. Maybe the entire tutorial should contain examples for when one makes sense over the other instead of listing reasons down here (I know there's an example above for an SQL transformation but its just a brief mention like "you could use that here").